packages/loader/container-loader/src/container.ts

```diff
@@ -778,7 +778,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             if (!hasAttachmentBlobs) {
                 // Get the document state post attach - possibly can just call attach but we need to change the
                 // semantics around what the attach means as far as async code goes.
-                const appSummary: ISummaryTree = this.context.createSummary();
+                const appSummary: ISummaryTree = await this.context.createSummary();
```

Comment:
Here is an example of the problem that I worry about when making createSummary() flow async. All capturing of state (aka revision) has to happen synchronously as otherwise we may not capture consistent state (i.e., parts of state would come from different points in time).
Here, we await container runtime summary creation, but capture protocol state and switch container to attaching state after that. That means that protocol state may change while we are blocked on await. Similar, any changes to container while we await would not generate ops because container is not switched to attaching state early enough, thus those changes would not make it to either snapshot nor ops that follow.

The only right way here is to ensure that all state is captured before any async activity is awaited. Ideally through 2-step process - get revision synchronously, generate snapshot from revision async. As a temp solution, we can do code inspection + UTs to get something working quickly (and fixing any cases like this one by ensuring that promise is captured early but await is pushed to line 793).  But this can't be long-term solution